### PR TITLE
202308-notebook-distribution

### PR DESCRIPTION
**Pull request by petermarshallio**
A new notebook on analysing data distribution, including estimating rank, quantiles, and histograms. This has a focus on the Quantiles sketch.

<hr>

This PR has:

- [x] been self-reviewed.
- [x] been tested in a test Druid cluster.
- [ ] been tested on the Jupyter docker container.

### CODE REVIEW

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+   "source": [
+    "## Estimating the rank (cumulative distribution functions)\n",
+    "\n",
+    "Druid provides two functions for estimating what the [normalized rank](https://datasketches.apache.org/docs/Quantiles/SketchingQuantilesAndRanksTutorial.html) of a value is within the data. Or, to put it another way, what percentage of other records fall below a given value.\n",
+    "\n",
+    "### Estimate a single rank position\n",
+    "\n",
+    "Use the [`DS_RANK`](https://druid.apache.org/docs/latest/querying/sql-functions#ds_rank) function to estimate where a given value falls in the distribution of the data, returned as a percentage.\n",
+    "\n",
+    "Run the following cell which returns the rank of a particular value in the \"XE\" flights on the 24th November 2005.\n",
```
Comment (vtlim) [inline examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:413]:
```suggestion
    "Run the following cell to obtain the rank of a particular value in the \"XE\" flights on the 24th November 2005.\n",
```

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 2]
```diff
@@ -0,0 +1,606 @@
...
+    "\n",
+    "display.sql(sql)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f06cb646-3ab2-4f7d-865c-dabc669390ce",
+   "metadata": {},
+   "source": [
+    "This tells us that this distance is representative of about 90% of flights on that day. The results for \"XE\" in the previous cell, where we showed the deciles for the same flights, corroborates that.\n",
```
Comment (vtlim) [inline examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:441]: What distance? 950? Are there units?

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 3]
```diff
@@ -0,0 +1,606 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "ea9b2ef0-b323-4b1c-b8dc-755e6fff32d7",
+   "metadata": {},
+   "source": [
+    "The results show an estimated count of the number of flights in each of the `Distance` buckets specified, plus one final bucket for the remainder.\n",
+    "\n",
+    "Combining `DS_QUANTILES_SKETCH` with `FILTER` focuses the distribution being analysed.\n",
+    "\n",
+    "Run the next cell which produces histograms for two specific `Reporting_Airline` with flights from Los Angeles."
```
Comment (vtlim) [inline examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:543]: For this cell and the previous one, could be cool to see them in histogram form: https://matplotlib.org/stable/gallery/statistics/hist.html#sphx-glr-gallery-statistics-hist-py

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 4]
```diff
@@ -0,0 +1,606 @@
...
+    "## Summary\n",
+    "\n",
+    "* Apache Datasketch \"Quantile\" sketches are supported in Apache Druid\n",
+    "* SQL functions exist for estimating individual ranks (`DS_RANK`) and quantiles (`DS_GET_QUANTILE`)\n",
+    "* Druid can return an `ARRAY` of ranks (`DS_CDF`) and quantiles (`DS_GET_QUANTILES`)\n",
+    "* Histograms can be estimated using the `DS_HISTOGRAM` function\n",
+    "\n",
+    "## Learn more\n",
+    "\n",
+    "* Read [quantiles extension](https://druid.apache.org/docs/latest/development/extensions-core/datasketches-quantiles) documentation\n",
```
Comment (vtlim) [inline examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:579]:
```suggestion
    "* Read the [quantiles extension](https://druid.apache.org/docs/latest/development/extensions-core/datasketches-quantiles) documentation\n",
```

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 5]
```diff
@@ -0,0 +1,606 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "4188e2c5-4ab0-45e3-9371-52d510a97413",
+   "metadata": {},
+   "source": [
+    "# Analysing data distributions\n",
```
Comment (vtlim) [inline examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:8]:
```suggestion
    "# Analyzing data distributions\n",
```

**Review by vtlim — CHANGES_REQUESTED** [review]
Great addition! Mostly minor comments on style and readability.

**Comment by vtlim** [comment]
Reminder to also add the Apache license to the notebook

File: examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb [hunk 6]
```diff
@@ -0,0 +1,606 @@
...
+  {
+   "cell_type": "markdown",
+   "id": "6f687b0a-517a-4df3-a7cb-b7095a11bb4b",
+   "metadata": {},
+   "source": [
+    "## Understanding the Quantiles sketch\n",
+    "\n",
+    "A structure known as a \"sketch\" holds a representation of the distribution of values that we will use throughout this notebook. These are created by using the [`DS_QUANTILES_SKETCH`](https://druid.apache.org/docs/latest/querying/sql-functions#ds_quantiles_sketch) function, providing it with the data set that we want to create the representation of.\n",
+    "\n",
+    "To see how quantiles sketches look, run the following cell. It uses `DS_QUANTILES_SKETCH` to create a Quantiles sketch for both `Distance` and `Delay`."
```
[the remark this answers is not in the document]
Comment (petermarshallio) [reply examples/quickstart/jupyter-notebooks/notebooks/03-query/03-dataDistribution.ipynb:151]: Thanks!

**Pull request closed without merging**